src/NUnitFramework/framework/Constraints/SubstringConstraint.cs

```diff
@@ -45,10 +52,18 @@ public SubstringConstraint(string expected) : base(expected)
         /// <returns>True for success, false for failure</returns>
         protected override bool Matches(string actual)
         {
-            if (this.caseInsensitive)
-                return actual != null && actual.ToLower().IndexOf(expected.ToLower()) >= 0;
-            else
-                return actual != null && actual.IndexOf(expected) >= 0;
+            return actual != null && actual.IndexOf(expected, comparisonType) >= 0;
+        }
+
+        /// <summary>
+        /// Flag the constraint to use the supplied StringComparison type.
+        /// </summary>
+        /// <param name="comparisonType ">The StringComparison type to be used.</param>
+        /// <returns>Self.</returns>
+        public SubstringConstraint Using(StringComparison comparisonType)
```

Comment:
This is a different use of Using from any others we have, but I think it should be clear to users.